capi/include/hyper.h

```diff
@@ -105,6 +107,9 @@ hyper_task *hyper_body_data(hyper_body *body);
  The `hyper_buf` pointer is only a borrowed reference, it cannot live outside
  the execution of the callback. You must make a copy to retain it.
 
+ The callback should return `HYPER_ITER_CONTINUE` to continue iterating
+ chunks as they are received, or `HYPER_ITER_BREAK` to cancel.
```

Comment:
Ah yes, this reminds me of another detail. What if my callback fails in a way that calls for erroring out completely, like out of memory? `HYPER_ITER_BREAK` aborts the iteration but `hyper_headers_foreach` doesn't return an error code. Do I need to set my own state variable in the struct I point `userdata` to, or can we have a dedicated return code that makes `hyper_headers_foreach` return an error code?

Reply:
What's the more common pattern? I'm fine with either option.

Reply:
As a user, it is slightly easier and more convenient to have a dedicated return code for **"error"** and have `hyper_headers_foreach()` be able to return success/error based on that. That'd be my preference. But if there's a reason not to do that, I can still deal with it without trouble.